realm/realm-library/src/main/java/io/realm/internal/OsSchemaInfo.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2017 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm.internal;
+
+import java.util.List;
+
+/**
+ * Java wrapper for the Object Store Schema object.
+ * <p>
+ * When it is created from java binding, it is used for initializing/validating the schemas through Object Store. It
+ * won't contain the column indices information.
+ * <p>
+ * When this is get from the Object Store {@code SharedRealm} instance, this represents the real schema of the Realm
+ * file. It will contain the all the schema information as well as the information about the column indices.
+ */
+public class OsSchemaInfo implements NativeObject {
+    private long nativePtr;
+    private static final long nativeFinalizerPtr = nativeGetFinalizerPtr();
+
+    /**
+     * Construct a {@code OsSchemaInfo} object from a given {@code OsObjectSchemaInfo} list.
+     *
+     * @param objectSchemaInfoList all the object schemas should be contained in this {@code OsObjectSchemaInfo}.
+     */
+    public OsSchemaInfo(List<OsObjectSchemaInfo> objectSchemaInfoList) {
+        long[] schemaNativePointers = new long[objectSchemaInfoList.size()];
+        for (int i = 0; i < objectSchemaInfoList.size(); i++) {
+            schemaNativePointers[i] = objectSchemaInfoList.get(i).getNativePtr();
+        }
+        this.nativePtr = nativeCreateFromList(schemaNativePointers);
+        NativeContext.dummyContext.addReference(this);
```

Comment:
Not sure why this is called `dummyContext`. It seems a rather weird name, but probably outside the scope of this PR